distribution/sc-proxy-dex/src/proxy_common.rs

```diff
@@ -31,6 +35,76 @@ pub trait ProxyCommonModule {
         Ok(())
     }
 
+    #[payable("*")]
+    #[endpoint]
+    fn acceptPay(
+        &self,
+        #[payment_token] token_id: TokenIdentifier,
+        #[payment] amount: Self::BigUint,
+    ) {
+        let tx_hash = self.blockchain().get_tx_hash();
```

Comment:
nice resolve - this function always uses farm and swap contracts in a synchronous mode - thus there is no way for someone to rewrite these value - like the last tx-hash. Although please add tests which validate in all cases.

Reply:
i did tests. they are on another branch and PR